doc/aaa/ldap/hld_ldap.md

```diff
@@ -0,0 +1,453 @@
+| :-------------------------------------------------------------------------------: | :-------------------------------------------------------: |
+|  https://wiki.debian.org/LDAP/NSS                                                 | Debian LDAP WIKI                                          |
+|  https://www.linuxbabe.com/debian/set-up-openldap-server-debian                   | Configure Ldap client in Debian 11                        |
+|  https://packages.debian.org/sid/libnss-ldapd                                     | libnss-ldapd                                              |
+|  https://tldp.org/HOWTO/archived/LDAP-Implementation-HOWTO/pamnss.html            | Implementation details of ldap pam & nss                  |
+|  https://wiki.archlinux.org/title/OpenLDAP                                        | slap & ldap-utils packages                                |
+|  https://packages.debian.org/sid/libpam-ldapd                                     | libpam-ldapd                                              |
+|  https://stackoverflow.com/questions/18756688/what-are-cn-ou-dc-in-an-ldap-search | DN, DC, OU, CN descriptions                               |
+|  https://manpages.debian.org/stretch/nslcd/nslcd.conf.5                           | NSLCD service configuration                               |
+
+
+### Overview 
+
+LDAP, the Lightweight Directory Access Protocol, is a mature, flexible, and well supported standards-based mechanism for interacting with directory servers. It’s often used for authentication and storing information about users, groups, and applications, but an LDAP directory server is a general-purpose data store and can be used in a wide variety of applications.
+In simple words, instead to config hundreds of same usernames and passwords in different switch devices, you can configure it one time and the validate in every case will be derive to the LDAP server instead locally in every device.
+https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol 
+
+LDAP motivation\
+![ldap_intro](ldap_intro.jpg)
+
+### Requirements
+
+-	Support LDAP login authentication for ssh/serial connection from an external LDAP server.
+-	Support to search for user/groups/etc in LDAP server from the switch LDAP client.
+-	Support a fallback mechanism for authentication. If an LDAP server authentication fails, the next server authentication will be performed or local authentication.
+-	TODO: SONIC rest API/nginx?
+
+### Architecture Design 
+
+Arc design diagram\
+![arc_ldap_sonic](ldap_sonic.jpg)
+
+### High-Level Design 
+
+#### hostcfgd
+Hostcfgd – listen to changes in CONFIG_DB in the LDAP table, and when the table has a new modification/or init happens it will trigger a callback in hostcfgd handle in AAA class to modify the PAM & NSS configuration files in Linux.
+
+#### SSH/Local login
+After enabling the LDAP configuration the ssh/local login to the switch will be authenticated by the LDAP server.
```

Comment:
I have one query related to the authentication. Is there a passkey used to locally authenticate an user against the centralised identity management system?

Reply:
yes - similat to tacacs/radius, you have a passkey to bind to the AAA server, and then, a user need to authenticate via the AAA server authentication defined in the AAA server.